source/server/admin/prometheus_stats.cc

```diff
@@ -274,6 +276,35 @@ uint64_t outputPrimitiveStatType(Buffer::Instance& response, const StatsParams&
   return result;
 }
 
+/*
+ * Returns the prometheus output for a summary. The output is a multi-line string (with embedded
+ * newlines) that contains all the individual quantile values and sum/count for a single histogram
+ * (metric_name plus all tags).
+ */
+std::string generateSummaryOutput(const Stats::ParentHistogram& histogram,
+                                  const std::string& prefixed_tag_extracted_name) {
+  const std::string tags = PrometheusStatsFormatter::formattedTags(histogram.tags());
+  const std::string hist_tags = histogram.tags().empty() ? EMPTY_STRING : (tags + ",");
+
+  const Stats::HistogramStatistics& stats = histogram.cumulativeStatistics();
```

Comment:
Given that a summary can't be aggregated in the same way a histogram can, would it make more sense to use `intervalStatistics()` here? https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#summary suggests that the last 5-10 minutes of data are preferable; we don't have any way to get that time window though with the collected data, at least not as a rolling window.

Reply:
Yeah, this is a bit unfortunate, as I would imagine getting a rolling window would be quite a bit more work...

Another approach could be to use tags to emit both the interval and cumulative summaries as the `quantileSummary` function does?

Reply:
I'm not sure if prometheus (the server/scraper) would ingest everything properly with two sets of quantiles, only different in tags. I'm not sure what the right thing to do here is. Agreed that getting the rolling window would be a lot more work.